juju/framework.py

```diff
@@ -189,13 +194,8 @@ def __init__(self, parent, key):
         # TODO This can probably be dropped, because the event type is only
         # really relevant if someone is either emitting the event or observing
         # it.
-        for cls in type(self).__mro__:
-            for attr_name, attr_value in cls.__dict__.items():
-                if isinstance(attr_value, Event):
-                    event_type = attr_value.event_type
-                    event_kind = attr_name
-                    emitter = self
-                    self.framework.register_type(event_type, emitter, event_kind)
+        for event_kind, unbound_event in inspect.getmembers(type(self), _is_unbound_event):
```

Comment:
Could just have `lambda obj: isinstance(obj, Event)`.

Reply:
Could.  It's used twice, so I thought it might be worth factoring it out, but I'm ok with inlining it.  Either way.